pageserver/benches/bench_layer_map.rs

```diff
@@ -254,5 +254,34 @@ fn bench_from_real_project(c: &mut Criterion) {
     });
 }
 
-criterion_group!(benches, bench_from_captest_env, bench_from_real_project);
+// Benchmark using synthetic data. Arrange image layers on stacked diagonal lines.
+fn bench_sequential(c: &mut Criterion) {
+    let mut layer_map = LayerMap::default();
+    for i in 0..100_000 {
+        // % 100 => 17 sec (very bad, should be way under 1 sec. 20x room for improvement)
+        // % 1000 => 1.6 sec
+        // % 10_000 =>
+        // TODO try inserting a super-wide layer in between every 10
+        let i32 = (i as u32) % 100;
+        let zero = Key::from_hex("000000000000000000000000000000000000").unwrap();
+        let layer = DummyImage {
+            key_range: zero.add(10 * i32)..zero.add(10 * i32 + 1),
+            lsn: Lsn(10 * i),
+        };
+        layer_map.insert_historic(Arc::new(layer));
+    }
+
+    let queries: Vec<(Key, Lsn)> = uniform_query_pattern(&layer_map);
+
+    // Test with uniform query pattern
+    c.bench_function("sequential_uniform_queries", |b| {
```

Comment:
In my case estimated time is 2577.3 s 
```
Benchmarking sequential_uniform_queries: Collecting 100 samples in estimated 2577.3 s (100 iterations)^C
```
It seems to be too large and actually not necessary.
May be add .step_by(100) to `uniform_query_pattern` ?

Reply:
Looks like search time greatly depends on key.
Without `step_by(100)` time is increased not 100, but 100 times:
```
Benchmarking sequential_uniform_queries: Collecting 100 samples in estimated 39sequential_uniform_queries
                        time:   [35.525 s 36.184 s 36.887 s]
                        change: [+14507% +14823% +15137%] (p = 0.00 < 0.05)
                        Performance has regressed.
Found 2 outliers among 100 measurements (2.00%)
  1 (1.00%) high mild
  1 (1.00%) high severe


real	64m34,645s
```

How it is possible to reduce number of samples?

Reply:
I'll randomly sample 100 out of 100_1000 queries with fixed seed